### PR TITLE
Add access log codes for fault injection

### PR DESCRIPTION
**Pull request by rshriram**
This PR adds an access log code "FI". FI is added to access log whenever a request is aborted by the fault injection filter.

### CODE REVIEW

File: include/envoy/http/access_log.h [hunk 1]
```diff
@@ -32,6 +32,8 @@ enum class FailureReason {
   UpstreamOverflow,
   // No route found for a given request.
   NoRouteFound,
+  // Used when faults (abort with error codes) are injected
```
Comment (RomanDzhabarov) [inline include/envoy/http/access_log.h]: minor: '.' at the very end of comment.

File: source/common/http/access_log/access_log_formatter.cc [hunk 2]
```diff
@@ -16,6 +16,7 @@ const std::string FilterReasonUtils::UPSTREAM_CONNECTION_FAILURE = "UF";
 const std::string FilterReasonUtils::UPSTREAM_CONNECTION_TERMINATION = "UC";
 const std::string FilterReasonUtils::UPSTREAM_OVERFLOW = "UO";
 const std::string FilterReasonUtils::NO_ROUTE_FOUND = "NR";
+const std::string FilterReasonUtils::FAULT_INJECTED = "FI";
 
 const std::string& FilterReasonUtils::toShortString(FailureReason failure_reason) {
```
Comment (RomanDzhabarov) [inline source/common/http/access_log/access_log_formatter.cc]: add test here: https://github.com/lyft/envoy/blob/master/test/common/http/access_log/access_log_formatter_test.cc#L14

File: source/common/http/filter/fault_filter.cc [hunk 3]
```diff
@@ -137,6 +137,7 @@ void FaultFilter::abortWithHTTPStatus() {
                                   "fault.http.abort.http_status", config_->abortCode()))}}};
   callbacks_->encodeHeaders(std::move(response_headers), true);
   config_->stats().aborts_injected_.inc();
+  callbacks_->requestInfo().onFailedResponse(Http::AccessLog::FailureReason::FaultInjected);
```
Comment (RomanDzhabarov) [inline source/common/http/filter/fault_filter.cc]: pls add test to fault_filter_test that onFailedResponse was called with the Http::AccessLog::FailureReason::FaultInjected arg.

**Comment by rshriram** [comment]
done

**Comment by RomanDzhabarov** [comment]
thanks, LGTM and sorry for delay (i'm OOO today)

**Comment by rshriram** [comment]
np. Thanks for the pointers on the test cases! that really helped.

**Pull request merged**